src/notifications/executeNotificationAction.ts

```diff
@@ -1,23 +1,44 @@
 import openHub from "../hub/openHub";
 import {
-  MessageActions,
+  MessageAction,
+  MessageActionsEnum,
   NOTIFICATIONS_OPEN_QUERY_PARAM,
+  OpenHubWithAction,
   StateType,
 } from "../globals/consts";
 
 export default async function executeNotificationAction(
-  selectedActions: MessageActions[] | undefined
+  selectedActions: MessageAction[] | undefined
 ): Promise<void> {
-  if (selectedActions?.includes(MessageActions.OPEN_HUB)) {
+  if (selectedActions?.includes(MessageActionsEnum.OPEN_HUB)) {
     return openHub(StateType.NOTIFICATION)();
   }
 
-  if (selectedActions?.includes(MessageActions.OPEN_NOTIFICATIONS_IN_HUB)) {
+  if (selectedActions?.includes(MessageActionsEnum.OPEN_NOTIFICATIONS_IN_HUB)) {
     return openHub(
       StateType.NOTIFICATION,
       `/home?${NOTIFICATIONS_OPEN_QUERY_PARAM}`
     )();
   }
 
+  const openHubWithActions = selectedActions
+    ?.map((action) => action as OpenHubWithAction)
```

Comment:
This seems like a bug? What if the message don't have the action? Can we actually filter by something and not only make TS happy?

Reply:
Make sure it has queryParams and path?

Reply:
It is just typescript, once we have this weird mixed array you need to actually check if the property exists on the object and that the way to identify it.